apps/web/src/components/conditions/Conditions.tsx

```diff
@@ -108,92 +119,100 @@ export function Conditions({
-                    onClick={() => handleDuplicate(index)}
-                    icon={<Duplicate />}
-                  >
-                    Duplicate
-                  </Dropdown.Item>
-                  <Dropdown.Item
-                    data-test-id="conditions-row-delete"
-                    onClick={() => handleDelete(index)}
-                    icon={<Trash />}
-                  >
-                    Delete
-                  </Dropdown.Item>
-                </Dropdown>
-              </Grid.Col>
-            </Grid>
-          </div>
-        );
-      })}
+                        </Wrapper>
+                      ) : (
+                        <Text ml={14} color={colors.B60}>
+                          When
+                        </Text>
+                      )}
+                    </Grid.Col>
+                    <Grid.Col span={3}>
+                      <Controller
+                        control={control}
+                        name={`conditions.0.children.${index}.on`}
+                        defaultValue={defaultOnFilter}
+                        render={({ field }) => {
+                          return (
+                            <Select
+                              placeholder="On"
+                              data={FilterPartTypeList}
+                              {...field}
+                              onChange={handleOnChildOnChange(index)}
+                              data-test-id="conditions-form-on"
+                            />
+                          );
+                        }}
+                      />
+                    </Grid.Col>
+                    <When truthy={filterFieldOn === FilterPartTypeEnum.WEBHOOK}>
+                      <WebHookUrlForm control={control} index={index} />
+                      <EqualityForm webhook control={control} index={index} />
+                    </When>
+                    <When truthy={filterFieldOn === FilterPartTypeEnum.PREVIOUS_STEP}>
+                      <PreviousStepsConditionRow customData={customData} control={control} index={index} />
+                    </When>
+                    <When
+                      truthy={[FilterPartTypeEnum.IS_ONLINE, FilterPartTypeEnum.IS_ONLINE_IN_LAST].includes(
+                        filterFieldOn
+                      )}
+                    >
+                      <OnlineConditionRow fieldOn={filterFieldOn} control={control} index={index} />
+                    </When>
+                    <When
+                      truthy={[
+                        FilterPartTypeEnum.PAYLOAD,
+                        FilterPartTypeEnum.SUBSCRIBER,
+                        FilterPartTypeEnum.TENANT,
+                      ].includes(filterFieldOn)}
+                    >
+                      <EqualityForm customData={customData} control={control} index={index} />
+                    </When>
+                  </Grid>
+                </Grid.Col>
```

Comment:
few suggestions to make it easier to understand:
- move the components from this file to separate files
- simplify the component body with the below suggestion, let's not care about the duplication

```
return (
   <div>
      <When truthy={filterFieldOn === FilterPartTypeEnum.WEBHOOK}>
         <WebhookRow ... /> // <--- contains all the inputs
      </When>
      <When truthy={filterFieldOn === FilterPartTypeEnum.PREVIOUS_STEP}>
         <PreviousStepRow ... /> // <--- contains all the inputs
      </When>
      <When truthy={[FilterPartTypeEnum.IS_ONLINE, FilterPartTypeEnum.IS_ONLINE_IN_LAST].includes(
           filterFieldOn
       )}>
         <OnlineConditionRow ... /> // <--- contains all the inputs
      </When>
      <When
            truthy={[FilterPartTypeEnum.PAYLOAD, FilterPartTypeEnum.SUBSCRIBER, FilterPartTypeEnum.TENANT].includes(filterFieldOn)}
        >
           <RegularConditionRow ... /> // <--- contains all the inputs
        </When>
   </div>
);
```